dlt/pipeline/pipeline.py

```diff
@@ -1247,13 +1271,15 @@ def _configure(
             )
 
         self.must_attach_to_local_pipeline = must_attach_to_local_pipeline
-        # attach to pipeline if folder exists and contains state
-        if has_state:
```

Comment:
`_get_prev_dev_mode_from_state` reproduces what `get_local_state_val` already does (it will load and migrate state). I think we can reuse it. We also need to "migrate" the state:
1. in `def _get_state(self)` please set `dev_mode` in local state key does not exist (see what we do with initial_cwd)
2. get previous dev mode with `get_local_state_val`
3. then we recreate pipeline or not - as you do below.